File: Segundo/Sistemas de gestión empresarial/002-Instalación y configuración de sistemas ERP-CRM/004-Parámetros de configuración del sistema ERP-CRM /101-Ejercicios/019-Mejoras/modules/var_get.py
# modules/var_get.py
TOOL = {
    "type": "var_get",
    "label": "Leer variable",
    "description": "Devuelve el contenido de una variable por nombre.",
    "config": {
        "name": {"type": "string", "label": "Nombre de variable", "default": "x"},
        "default": {"type": "string", "label": "Valor por defecto", "default": ""}
    }
}

def execute(config, context):
    name = (config.get("name") or "x").strip()
    default = config.get("default", "")
    vars_ = context.get("vars", {})
    return {"name": name, "value": vars_.get(name, default)}

File: Segundo/Sistemas de gestión empresarial/002-Instalación y configuración de sistemas ERP-CRM/004-Parámetros de configuración del sistema ERP-CRM /101-Ejercicios/019-Mejoras/static/modules/var_get.js
export default {
  type: "var_get",
  buildBody(el, toolSpec, nodeId){
    const body = el.querySelector(".body");
    const defN = toolSpec?.config?.name?.default ?? "x";
    const defD = toolSpec?.config?.default?.default ?? "";
    body.innerHTML = `
      <div class="form-row">
        <label for="name_${nodeId}">Nombre de variable</label>
        <input id="name_${nodeId}" type="text" value="${defN}" placeholder="x">
      </div>
      <div class="form-row">
        <label for="default_${nodeId}">Valor por defecto</label>
        <input id="default_${nodeId}" type="text" value="${defD}" placeholder="">
      </div>
    `;
  },
  readConfig(el){
    const inputs = el.querySelectorAll(".body input");
    const name = inputs?.[0]?.value?.trim() || "x";
    const def  = inputs?.[1]?.value ?? "";
    return { name, default: def };
  },
  renderResult(el, data){
    const body = el.querySelector(".body");
    const d = document.createElement("div");
    d.style.font="12px ui-monospace";
    d.style.marginTop="6px";
    d.textContent = `→ ${data?.name}: ${String(data?.value)}`;
    body.appendChild(d);
  }
};

